lib/api/objectCopy.js

```diff
@@ -351,6 +356,29 @@ function objectCopy(authInfo, request, sourceBucket,
                     serverSideEncryption, destBucketMD);
             }
 
+            if (storeMetadataParams.locationMatch && locationType &&
```

Comment:
I don't think you need to check if locationType exists because if it doesn't the next condition just won't be met, right?